x/rollup/types/adapters_test.go

```diff
@@ -0,0 +1,303 @@
+		require.NoError(t, err)
+		require.Empty(t, cosmosTxs)
+	})
+
+	src := rand.NewSource(0)
+	r := rand.New(src)
+
+	t.Run("non-zero txs without error", func(t *testing.T) {
+		testTable := []struct {
+			name              string
+			depNum, nonDepNum int
+		}{
+			{
+				name:   "DepositTx",
+				depNum: 1,
+			},
+			{
+				name:   "Multiple DepositTxs",
+				depNum: 3,
+			},
+			{
+				name:      "DepositTx + AccessListTx",
+				depNum:    1,
+				nonDepNum: 1,
+			},
+			{
+				name:      "Multiple DepositTxs + DynamicFeeTxs",
+				depNum:    3,
+				nonDepNum: 3,
+			},
+		}
+
+		interfaceRegistry := codectypes.NewInterfaceRegistry()
+		rollupv1.RegisterInterfaces(interfaceRegistry)
+		protoCodec := codec.NewProtoCodec(interfaceRegistry)
+
+		for _, tc := range testTable {
+			t.Run(tc.name, func(t *testing.T) {
+				ethTxs := make([]hexutil.Bytes, tc.depNum+tc.nonDepNum)
+				transactions := generateEthTransactions(t, tc.depNum, tc.nonDepNum)
+
+				for i := range transactions {
+					txBinary, err := transactions[i].MarshalBinary()
+					require.NoError(t, err)
+					ethTxs[i] = txBinary
+				}
+
+				// Convert the binary format to a Cosmos transaction.
+				cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs(ethTxs)
+				require.NoError(t, err)
+
+				if len(ethTxs) == 0 {
+					require.Equal(t, 0, len(cosmosTxs))
+					return
+				}
+
+				var decodedTx sdktx.Tx
+				err = decodedTx.Unmarshal(cosmosTxs[0])
+				require.NoError(t, err)
+
+				var applyL1TxsRequest rollupv1.ApplyL1TxsRequest
+				err = protoCodec.Unmarshal(decodedTx.GetBody().GetMessages()[0].GetValue(), &applyL1TxsRequest)
+				require.NoError(t, err)
+
+				// Copy the original transaction because time fields are different if not copied.
+				require.Equal(t, tc.depNum, len(applyL1TxsRequest.TxBytes))
+				for i, txBytes := range applyL1TxsRequest.TxBytes {
+					newTransaction := transactions[i]
+					err = newTransaction.UnmarshalBinary(txBytes)
+					require.NoError(t, err)
+					require.Equal(t, transactions[i], newTransaction)
+				}
+
+				for i := 1; i < len(cosmosTxs); i++ {
+					require.Equal(t, transactions[tc.depNum-1+i].Data(), []byte(cosmosTxs[i]))
+				}
+			})
+		}
+	})
+
+	t.Run("non-zero txs with error", func(t *testing.T) {
+		t.Run("unmarshal binary error", func(t *testing.T) {
+			cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{[]byte("invalid")})
+			require.Nil(t, cosmosTxs)
+			require.Error(t, err)
+		})
+		t.Run("zero deposit txs", func(t *testing.T) {
+			inner := generateDynamicFeeInner(r)
+			transaction := ethtypes.NewTx(inner)
+			txBytes, err := transaction.MarshalBinary()
+			require.NoError(t, err)
+			cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{txBytes})
+			require.Nil(t, cosmosTxs)
+			require.Error(t, err)
+		})
+		t.Run("NewAnyWithValue error", func(t *testing.T) {
+			t.Skip()
+			// TODO: Implement this test case
```

Comment:
Is this PR ready for review?

Reply:
Yes, I am just not really sure if we really need to implement these test cases.